### DISCUSSION TITLE
Exclude arbitrary texts inside components from index?

### DISCUSSION THREAD

**Discussion by beyond9thousand** [retypeapp/retype#443] [Q&A]
Is it possible to exclude the icon name, badge content etc that may be suffixed to panel names from index reference?

Example: https://beyond9thousand.github.io/meta/guides/privacy-tools/#two-factor-authentication-apps

vs

https://beyond9thousand.github.io/meta/guides/software/#aniyomi-badge-variant-info-text-anime-badge-variant-info-text-manga

As you have probably identified the problem here, it packs too much unnecessary information from text that is meant to serve other purposes (like displaying icons, badges etc)

**Comment by geoffreymcgill** [comment]
Can you provide some more details regarding this scenario as I was unable to reproduce? 

Neither search query I tried included the icon or badge in the search results. 

<img width="796" alt="Screenshot 2022-11-21 at 2 59 19 PM" src="https://user-images.githubusercontent.com/62210/203073918-fa5eec1f-680c-4be3-9c1d-8209beea530a.png">
<img width="745" alt="Screenshot 2022-11-21 at 2 58 52 PM" src="https://user-images.githubusercontent.com/62210/203073930-c6975c15-ba22-44ec-a5b4-6b1793edc5d9.png">

What search query should I be using to reproduce what you are seeing?

**Comment by beyond9thousand** [comment]
I should rephrase it, you notice the urls that I shared? notice how the name includes the `icon-name` and `badge-variant-etc...`
that is what i am referring to. conveniently (good thing) this doesn't happen up to sub heading level 3, anything after this grabs everything from that line in the ref link

![image](https://user-images.githubusercontent.com/77363063/203132340-886baf8a-1f7b-45a1-9afa-6cc9f6f3d868.png)
![image](https://user-images.githubusercontent.com/77363063/203132428-93e87acc-29f7-4a6c-bf48-f5626e4e1202.png)

**Reply by geoffreymcgill** [reply]
okay, I see now. Yes, that is a defect. Those anchors should only include raw characters.

I will create an issue and we will fix.

**Reply by beyond9thousand** [reply]
Ah TIL these are called anchors. Thanks for taking the time to cross-check this. Looking forward to the update.
